src/test-run/index.js

```diff
@@ -270,13 +272,43 @@ export default class TestRun extends Session {
         return this.currentDriverTask ? this.currentDriverTask.command : null;
     }
 
+    static _transformResizeWindowToFitDeviceCommand (command) {
+        var { landscapeWidth, portraitWidth } = getViewportSize(command.device);
+        var portrait = command.options.portraitOrientation;
+        var width    = portrait ? portraitWidth : landscapeWidth;
+        var height   = portrait ? landscapeWidth : portraitWidth;
+
+        return new ResizeWindowCommand({ width, height });
+    }
+
+    _isValidResize (command) {
```

Comment:
Event better move this function to `browserConnection` and name it `canResizeWindowToDimensions (width, height)`